### PR TITLE
[JENKINS-52664] Do refactor and support yml config file

### PR DESCRIPTION
**Pull request by pvtuan10**
JIRA issue: [JENKINS-52664](https://issues.jenkins-ci.org/browse/JENKINS-52664)

This is a part of the epic [Pipeline as YAML Incubator Plugin 1.0-rc](https://issues.jenkins-ci.org/browse/JENKINS-53713) to prepare for 1.0-rc release of the plugin

- [x] Support both format Jenkinsfile.yml and Jenkinsfile.yaml
- [x] Do code refactor, move files to packages

@oleg-nenashev @martinda

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -150,9 +144,6 @@
       </developer>
     </developers>
 
-    <!-- Assuming you want to host on @jenkinsci: -->
-    <url>https://wiki.jenkins.io/display/JENKINS/TODO+Plugin</url>
```
Comment (oleg-nenashev) [inline pom.xml]: I would create a Wiki stub. Until posting of docs from GitHub is supported, this is the best way to share info.

**Comment by pvtuan10** [comment]
I cannot merge this PR, I need to have committer access to merge.

File: pom.xml (hunk 1 above)
Comment (gautamabhishek46) [reply pom.xml]: :+1:

**Comment by gautamabhishek46** [comment]
@pvtuan10 Should I merge it ?

**Comment by gautamabhishek46** [comment]
I don't think that I can do that. I will check fir that for sure.

CC @oleg-nenashev

**Pull request merged**